terraform/modules/kubernetes/worker_app.tf

```diff
@@ -0,0 +1,83 @@
+resource "kubernetes_deployment" "main_worker" {
+  for_each = var.worker_apps
+
+  metadata {
+    name      = "${var.service_name}-${each.key}-${var.app_environment}"
+    namespace = var.namespace
+  }
+  spec {
+    replicas = each.value.replicas
+    selector {
+      match_labels = {
+        app = "${var.service_name}-${each.key}-${var.app_environment}"
+      }
+    }
+    strategy {
+      type = "Recreate"
+    }
+    template {
+      metadata {
+        labels = {
+          app = "${var.service_name}-${each.key}-${var.app_environment}"
+        }
+      }
+      spec {
+        node_selector = {
+          "teacherservices.cloud/node_pool" = "applications"
+          "kubernetes.io/os"                = "linux"
+        }
+        container {
+          name    = "${var.service_name}-${each.key}-${var.app_environment}"
+          image   = var.app_docker_image
+          command = try(slice( each.value.startup_command, 0, 1 ), null)
+          args    = try(slice( each.value.startup_command, 1, length(each.value.startup_command) ), null)
+
+          # Check performed to ensure the application is available. If it fails the current pod is killed and a new one created.
+          dynamic "liveness_probe" {
```

Comment:
As discussed, health probes for worker apps add little value but increase complexity so we should remove them